src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -12,6 +12,55 @@ namespace System.Text.Json.Tests
 {
     public static partial class Utf8JsonReaderTests
     {
+        // Test string, Argument to DateTime(Offset).Parse(Exact)
+        public static IEnumerable<object[]> ValidISO8601Tests()
+        {
+            yield return new object[] { "\"0997-07-16\"", "0997-07-16" };
+            yield return new object[] { "\"1997-07-16\"", "1997-07-16" };
+            yield return new object[] { "\"1997-07-16T19:20\"", "1997-07-16T19:20" };
+            yield return new object[] { "\"1997-07-16T19:20:30\"", "1997-07-16T19:20:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.45\"", "1997-07-16T19:20:30.45" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555\"", "1997-07-16T19:20:30.4555555" };
```

Comment:
I would add tests for 0 to 9 number of fractional seconds. Also, `.0, and `.0000...`